runtime/service.go

```diff
@@ -598,44 +600,63 @@ func (s *service) startVM(ctx context.Context, request *taskAPI.CreateTaskReques
 	}
 
 	cfg := firecracker.Config{
-		BinPath:         s.config.FirecrackerBinaryPath,
 		SocketPath:      s.config.SocketPath,
 		VsockDevices:    []firecracker.VsockDevice{{Path: "root", CID: cid}},
 		KernelImagePath: s.config.KernelImagePath,
 		KernelArgs:      s.config.KernelArgs,
-		RootDrive:       firecracker.BlockDevice{HostPath: s.config.RootDrive, Mode: "rw"},
-		CPUCount:        int64(s.config.CPUCount),
-		CPUTemplate:     firecracker.CPUTemplate(s.config.CPUTemplate),
-		MemInMiB:        256,
-		Console:         s.config.Console,
-		LogFifo:         s.config.LogFifo,
-		LogLevel:        s.config.LogLevel,
-		MetricsFifo:     s.config.MetricsFifo,
-		Debug:           s.config.Debug,
-	}
+		MachineCfg: models.MachineConfiguration{
+			VcpuCount:   int64(s.config.CPUCount),
+			CPUTemplate: models.CPUTemplate(s.config.CPUTemplate),
+			MemSizeMib:  256,
+		},
+		LogFifo:     s.config.LogFifo,
+		LogLevel:    s.config.LogLevel,
+		MetricsFifo: s.config.MetricsFifo,
+		Debug:       s.config.Debug,
+	}
+
+	idx := strconv.Itoa(1)
```

Comment:
Out of curiosity why isn't this just `idx := "1"`? Im guessing `"0"` is reserved for something else?

Reply:
Well that's certainly interesting. I can't find where that came from. There's _no way_ I'd have made up anything as insane as a 1-indexed array, so it must have been somewhere. But afaict the drive IDs have never even needed to be integers, let alone array indexes.

Let's fix that in a future PR.